test/framework/src/main/java/org/elasticsearch/test/MockUtils.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test;
+
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockingDetails;
+import static org.mockito.Mockito.when;
+
+/**
+ * Utilities for setting up Mockito mocks.
+ */
+public class MockUtils {
+
+    /**
+     * Sets up a mock TransportService that can answer calls to TransportService.getThreadPool().executor(String).
+     *
+     * @return A mocked TransportService instance
+     */
+    public static TransportService setupTransportServiceWithThreadpoolExecutor() {
+        TransportService transportService = mock(TransportService.class);
+        ThreadPool threadPool = mock(ThreadPool.class);
+        when(transportService.getThreadPool()).thenReturn(threadPool);
+        when(threadPool.executor(anyString())).thenReturn(EsExecutors.DIRECT_EXECUTOR_SERVICE);
+        return transportService;
+    }
+
+    /**
+     * Sets up a mock TransportService that can answer calls to TransportService.getThreadPool().executor(String), using the given
+     * threadPool in TransportService.
+     *
+     * @param threadPool A mock ThreadPool
+     * @return A mocked TransportService instance
+     */
+    public static TransportService setupTransportServiceWithThreadpoolExecutor(ThreadPool threadPool) {
+        assert mockingDetails(threadPool).isMock();
+
+        TransportService transportService = mock(TransportService.class);
+        when(transportService.getThreadPool()).thenReturn(threadPool);
+        when(threadPool.executor(anyString())).thenReturn(EsExecutors.DIRECT_EXECUTOR_SERVICE);
```

Comment:
Suspect we should probably have a method that takes a mock transport service and a mock threadpool and does these things, rather than duplicating the lines in the three methods.

Also we should probably mock `threadPool.generic()` to return the same executor.

Reply:
> Though [generic()'s implementation does fall through to executor()](https://github.com/elastic/elasticsearch/blob/aa3f0b6bd6483690ecdb9968af62e072f0cc2084/server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java#L431-L433), so I believe it is already effectively mocked.

I don't think that's how mocks work unfortunately, it'll return `null` rather than executing the real implementation. At least, it'd be worth confirming that.

The reasons for having a special `generic()` accessor are lost in the mists of time. It's probably a bad idea these days. But still, while it exists we should probably avoid making these mocks trappier than they need to be.

Reply:
> I don't think that's how mocks work unfortunately, it'll return null rather than executing the real implementation. At least, it'd be worth confirming that.

Ah, you're right, what I said doesn't make any sense. Mocks just have `null` defaults as you say.

> The reasons for having a special generic() accessor are lost in the mists of time. It's probably a bad idea these days. But still, while it exists we should probably avoid making these mocks trappier than they need to be.

Gotcha. I won't update the code to add new `generic()` calls, then.

Reply:
:+1: LGTM still